src/urllib3/util/ssl_.py

```diff
@@ -44,12 +44,11 @@ def _const_compare_digest_backport(a, b):
 # Borrow rfc3986's regular expressions for IPv4
 # and IPv6 addresses for use in is_ipaddress()
 _IP_ADDRESS_REGEX = re.compile(
-    r'^(?:%s|%s|%s|%s|%s)$' % (
+    r'^(?:%s|%s|%s|%s)$' % (
         abnf_regexp.IPv4_RE,
         abnf_regexp.IPv6_RE,
         abnf_regexp.IPv6_ADDRZ_RE,
-        abnf_regexp.IPv_FUTURE_RE,
-        abnf_regexp.IP_LITERAL_RE
+        abnf_regexp.IPv_FUTURE_RE
```

Comment:
I wonder if rfc3986 could just provide a regular expression for use here. :woman_shrugging:

Reply:
Probably, something like `misc.IP_ADDRESS_MATCHER`?

Reply:
I don't know that rfc3986 will ever need to use the matcher so I'd prefer not to add more time compiling a regexp we're not going to use. We could just have them all in one RegExp though in abnf_regexp

Reply:
That's true, I can create that issue to track the idea.